src/Context/ProjectContext.tsx

```diff
@@ -0,0 +1,65 @@
+
+import { IContextProps } from "@/Interfaces";
+import baseUrl from "@/utils/Custom/Custom";
+import { createContext, useContext, useState } from "react";
+import { ToastContext } from "./ToastContext";
+
+
+interface ProjectContextTypes {
+  handleGetData: (type: string, id: any,pages:number) => any
+  tableData: any
+
+}
+
+export const ProjectContext = createContext<ProjectContextTypes | null>(null);
+
+
+
+const ProjectContextProvider = ({ children }: IContextProps) => {
+
+  const toast = useContext(ToastContext)
+  let headers = {
+    Authorization: `Bearer ${localStorage.getItem('adminToken')}`,
+  }
+
+  const [tableData, setTableData] = useState(null)
+
+  // const [Loading, setLoading] = useState(false)
+  const handleGetData = (type: string, id: any,pages:number) => {
```

Comment:
you can set the type to only get and delete and provide  a default parameters to the function
 ` const handleGetData = (type: "delete" | "get" =  "get", id?: number ,pages:number = 5) => {...)`

Reply:
the function name is also here misleading. We are calling the function `handleGetData`, however, this function can get data and delete them as well.

Reply:
We can also improve the function by taking an object instead of single parameters, this makes the user write them in any order. as follows:
```
interface GetDataOptions {
  type?: "delete" | "get";
  id?: number;
  pages?: number;
}

const handleGetData = ({ type = "get", id, pages = 5 }: GetDataOptions) => {...}

// Omitting 'id' and 'pages', so default values will be used
handleGetData({ type: "get" });

// Providing values for 'id' and 'pages'
handleGetData({ type: "delete", id: 42, pages: 10 });
```